### PR TITLE
docs: reference for developers

### PR DESCRIPTION
**Pull request by danielhochman**
Going through this is exercise left me feeling that we could use more improvement to the structure of documentation. I think a developer FAQ might be useful. Moreso I think that we should move away from GitHub markdown docs and by rule keep everything in github.io docs. This adds some friction to writing new docs but makes the user experience a lot better (search and more robust indexing). Where appropriate we could leave markdown files in the expected places with a link to the corresponding .io doc.

Anyone have any thoughts on this?

### CODE REVIEW

**Comment by mattklein123** [comment]
I think we need to balance ease of update with ease of use. I could go either way, but IMO there will be less friction for dev docs being in markdown, and the volume of docs is not huge. I think this change is a great start, but ultimately I would like to see us flesh out DEVELOPER.md more with some more description of the basic flows with links to detail pages. I think this is a fine start though.

**Comment by htuch** [comment]
I would vote for keeping developer docs in GH MD FWIW, lowering the activation energy for modifying/adding docs will mean more docs.

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Developer documentation
```
Comment (htuch) [inline DEVELOPMENT.md]: DEVELOPMENT.md or DEVELOPER.md?

Comment (danielhochman) [inline DEVELOPMENT.md]: i've seen both in open source. i'll go with DEVELOPER though.

**Pull request merged**